pkg/batch/scan.go

```diff
@@ -22,18 +23,55 @@ type Options struct {
 type Batch struct {
 	Data     net.Buffers
 	Location Location
+
+	// backup hold the same data as Data. It is used to rewind if something goes wrong
+	// Because it copies the slice, the memory is not duplicated
+	// Because we only read data, the underlaying memory is not modified neither
```

Comment:
It has a test so I can prove that it works. So I think we can trust the implementation unless we can modify the test to make it fail.

The first [] is duplicated so it can keep the reference while the original one is modified. But the [] stored inside are just appended. This means the underlying array remains the same AFAIK.

Reply:
sorry I wasn't trying to say it doesn't work. But I think it's not true it doesn't duplicate any memory. I think append allocates memory for you for pointers in the first dimension of the matrix (line 44)